sonar-python-plugin/src/main/java/org/sonar/plugins/python/ruff/RuffJsonReportReader.java

```diff
@@ -81,22 +81,22 @@ private void onResult(JSONObject result) {
     issue.startLocationCol = toInteger(location.get("column"));
     issue.startLocationRow = toInteger(location.get("row"));
     JSONObject endLocation = (JSONObject) result.get("end_location");
-    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol);
     issue.endLocationRow = toInteger(endLocation.get("row"));
+    issue.endLocationCol = correctEndLocationCol(endLocation.get("column"), issue.startLocationCol, issue.startLocationRow, issue.endLocationRow);
     consumer.accept(issue);
   }
 
   /*
     Ruff returns the col number of the last char + 1.
     In order to properly read the col number we need to return the number of the last char.
    */
-  private static Integer correctEndLocationCol(Object value, int startLocationCol) {
+  private static Integer correctEndLocationCol(Object value, int startLocationCol, int startLocationRow, int endLocationRow) {
     Integer endLocationCol = toInteger(value);
     if (endLocationCol != null) {
-      if (endLocationCol > startLocationCol) {
-        return endLocationCol - 1;
-      } else {
+      if (endLocationRow == startLocationRow && endLocationCol <= startLocationCol) {
         return startLocationCol;
+      } else {
+        return endLocationCol -1 ;
```

Comment:
Nitpick: Maybe we should have a space between the `-` and the `1`, i.e.: `endLocationCol - 1`.